{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/3q/37b0tth55xxdfn82hcm7ylp80000gn/T/ipykernel_47742/625248955.py:4: UserWarning: Parsing dates in %d/%m/%Y format when dayfirst=False (the default) was specified. Pass `dayfirst=True` or specify a format to silence this warning.\n",
      "  orders['Order Date'] = pd.to_datetime(orders['Order Date'])\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "orders = pd.read_csv('./data/orders.csv')\n",
    "orders['Order Date'] = pd.to_datetime(orders['Order Date'])\n",
    "furniture = pd.Series(orders[orders.Category == 'Furniture'].groupby('Order Date').agg('sum')['Quantity']).asfreq(\"W\").fillna(0)\n",
    "office = pd.Series(orders[orders.Category == 'Office Supplies'].groupby('Order Date').agg('sum')['Quantity']).asfreq(\"W\").fillna(0).rename({\"Quantity\": \"Office\"})\n",
    "technology = pd.Series(orders[orders.Category == 'Technology'].groupby('Order Date').agg('sum')['Quantity'])\n",
    "technology['2014-01-01'] = 0\n",
    "technology = technology.asfreq(\"W\").fillna(0).rename({\"Quantity\": \"Technology\"})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "furniture.rename({\"Quantity\": \"Furniture\"}, inplace=True)\n",
    "data = pd.concat([office, furniture, technology], axis=1)\n",
    "data.columns = ['Office', \"Furniture\", \"Technology\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(205, 3)"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = data.iloc[:205]\n",
    "data.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 0-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 1-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 2-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 3-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 4-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 5-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 6-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 7-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 8-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 9-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 10-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 11-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 12-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 13-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 14-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 15-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 16-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 17-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 18-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 19-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 20-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 21-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 22-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 23-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 24-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 25-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 26-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 27-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 28-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 29-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 30-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 31-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 32-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 33-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 34-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 35-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 36-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 37-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 38-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 39-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 40-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 41-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 42-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 43-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 44-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 45-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 46-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 47-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 48-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 49-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 50-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 51-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 52-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 53-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 54-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 55-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 56-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 57-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 58-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 59-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 60-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 61-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 62-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 63-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 64-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 65-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 66-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 67-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 68-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 69-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 70-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 71-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 72-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 73-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 74-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 75-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 76-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 77-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 78-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 79-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 80-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 81-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 82-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 83-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 84-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 85-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 86-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 87-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 88-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 89-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 90-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 91-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 92-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 93-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 94-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 95-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 96-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 97-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 98-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 99-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 100-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 101-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 102-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 103-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 104-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 105-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 106-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 107-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 108-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 109-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 110-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 111-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 112-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 113-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 114-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 115-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 116-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 117-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 118-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 119-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 120-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 121-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 122-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 123-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 124-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 125-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 126-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 127-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 128-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 129-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 130-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 131-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 132-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 133-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 134-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 135-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 136-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 137-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 138-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 139-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 140-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 141-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 142-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 143-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 144-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 145-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 146-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 147-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 148-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 149-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 150-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 151-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 152-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 153-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n",
      "/Users/grigorychaykovsky/retail_forecasts/.venv/lib/python3.12/site-packages/sktime/utils/parallel.py:92: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster VAR failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message.\n",
      "                Fit failed for the 154-th data split, on training data y_train with\n",
      "                cutoff <NA>, and len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: VAR(maxlags=0).\n",
      "                \n",
      "  ret = [fun(x, meta=meta) for x in iter]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best Parameters: {'maxlags': 7}\n",
      "                Office  Furniture  Technology\n",
      "2000-07-24  198.699155  58.569227   62.522975\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "from sktime.forecasting.var import VAR\n",
    "from sktime.forecasting.model_selection import (\n",
    "    ForecastingGridSearchCV,\n",
    "    SlidingWindowSplitter\n",
    ")\n",
    "from sktime.performance_metrics.forecasting import mean_absolute_percentage_error\n",
    "\n",
    "# Ensure the index is a datetime index\n",
    "data.index = pd.date_range(start=\"2000-01-01\", periods=len(data), freq=\"D\")\n",
    "\n",
    "# Define 1-step ahead forecasting horizon\n",
    "fh = [1]\n",
    "\n",
    "# Define Sliding Window Cross-Validation\n",
    "cv = SlidingWindowSplitter(\n",
    "    window_length=50,  # Train on past 50 observations\n",
    "    fh=fh,\n",
    "    step_length=1  # Move the window 1 step forward\n",
    ")\n",
    "\n",
    "# Initialize VAR model\n",
    "forecaster = VAR()\n",
    "\n",
    "# Define hyperparameter grid\n",
    "param_grid = {\n",
    "    \"maxlags\": list(range(15))  # Lag order\n",
    "}\n",
    "\n",
    "# Grid Search with MAPE as the scoring metric\n",
    "grid_searcher = ForecastingGridSearchCV(\n",
    "    forecaster=forecaster,\n",
    "    param_grid=param_grid,\n",
    "    cv=cv,\n",
    "    scoring=mean_absolute_percentage_error\n",
    ")\n",
    "\n",
    "# Fit the model\n",
    "grid_searcher.fit(data)\n",
    "print(\"Best Parameters:\", grid_searcher.best_params_)\n",
    "\n",
    "# Make 1-step ahead prediction using the best model\n",
    "best_forecaster = grid_searcher.best_forecaster_\n",
    "y_pred = best_forecaster.predict(fh)\n",
    "print(y_pred)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'maxlags': 7}"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "grid_searcher.best_params_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "  Summary of Regression Results   \n",
       "==================================\n",
       "Model:                         VAR\n",
       "Method:                        OLS\n",
       "Date:           Sun, 02, Mar, 2025\n",
       "Time:                     20:36:13\n",
       "--------------------------------------------------------------------\n",
       "No. of Equations:         3.00000    BIC:                    19.0104\n",
       "Nobs:                     198.000    HQIC:                   18.3580\n",
       "Log likelihood:          -2550.37    FPE:                6.04372e+07\n",
       "AIC:                      17.9144    Det(Omega_mle):     4.40587e+07\n",
       "--------------------------------------------------------------------\n",
       "Results for equation Office\n",
       "================================================================================\n",
       "                   coefficient       std. error           t-stat            prob\n",
       "--------------------------------------------------------------------------------\n",
       "const                18.648877         8.669614            2.151           0.031\n",
       "L1.Office             0.270562         0.096915            2.792           0.005\n",
       "L1.Furniture          0.038766         0.233169            0.166           0.868\n",
       "L1.Technology         0.662083         0.239281            2.767           0.006\n",
       "L2.Office             0.005786         0.098822            0.059           0.953\n",
       "L2.Furniture          0.162416         0.232866            0.697           0.486\n",
       "L2.Technology         0.235294         0.246422            0.955           0.340\n",
       "L3.Office             0.293332         0.099031            2.962           0.003\n",
       "L3.Furniture         -0.280134         0.232543           -1.205           0.228\n",
       "L3.Technology        -0.249366         0.238599           -1.045           0.296\n",
       "L4.Office             0.024181         0.099436            0.243           0.808\n",
       "L4.Furniture          0.139718         0.233415            0.599           0.549\n",
       "L4.Technology        -0.592429         0.239624           -2.472           0.013\n",
       "L5.Office             0.009476         0.098898            0.096           0.924\n",
       "L5.Furniture          0.417766         0.232488            1.797           0.072\n",
       "L5.Technology        -0.140760         0.242419           -0.581           0.561\n",
       "L6.Office             0.062626         0.100277            0.625           0.532\n",
       "L6.Furniture         -0.088694         0.236882           -0.374           0.708\n",
       "L6.Technology        -0.122455         0.244518           -0.501           0.617\n",
       "L7.Office             0.049709         0.097103            0.512           0.609\n",
       "L7.Furniture          0.115423         0.240825            0.479           0.632\n",
       "L7.Technology         0.012587         0.242159            0.052           0.959\n",
       "================================================================================\n",
       "\n",
       "Results for equation Furniture\n",
       "================================================================================\n",
       "                   coefficient       std. error           t-stat            prob\n",
       "--------------------------------------------------------------------------------\n",
       "const                 7.594539         3.476941            2.184           0.029\n",
       "L1.Office             0.039394         0.038868            1.014           0.311\n",
       "L1.Furniture          0.089780         0.093512            0.960           0.337\n",
       "L1.Technology         0.091136         0.095963            0.950           0.342\n",
       "L2.Office            -0.000196         0.039633           -0.005           0.996\n",
       "L2.Furniture          0.097033         0.093391            1.039           0.299\n",
       "L2.Technology         0.020351         0.098827            0.206           0.837\n",
       "L3.Office             0.060184         0.039716            1.515           0.130\n",
       "L3.Furniture          0.040477         0.093261            0.434           0.664\n",
       "L3.Technology        -0.016113         0.095690           -0.168           0.866\n",
       "L4.Office             0.037641         0.039879            0.944           0.345\n",
       "L4.Furniture         -0.066471         0.093611           -0.710           0.478\n",
       "L4.Technology        -0.043038         0.096101           -0.448           0.654\n",
       "L5.Office             0.000753         0.039663            0.019           0.985\n",
       "L5.Furniture          0.138989         0.093239            1.491           0.136\n",
       "L5.Technology        -0.121959         0.097222           -1.254           0.210\n",
       "L6.Office             0.042453         0.040216            1.056           0.291\n",
       "L6.Furniture          0.023089         0.095001            0.243           0.808\n",
       "L6.Technology        -0.018700         0.098064           -0.191           0.849\n",
       "L7.Office             0.007659         0.038943            0.197           0.844\n",
       "L7.Furniture          0.054808         0.096582            0.567           0.570\n",
       "L7.Technology        -0.121098         0.097118           -1.247           0.212\n",
       "================================================================================\n",
       "\n",
       "Results for equation Technology\n",
       "================================================================================\n",
       "                   coefficient       std. error           t-stat            prob\n",
       "--------------------------------------------------------------------------------\n",
       "const                 3.811902         3.302156            1.154           0.248\n",
       "L1.Office             0.041570         0.036914            1.126           0.260\n",
       "L1.Furniture          0.053799         0.088811            0.606           0.545\n",
       "L1.Technology         0.228832         0.091139            2.511           0.012\n",
       "L2.Office             0.024481         0.037640            0.650           0.515\n",
       "L2.Furniture          0.014545         0.088696            0.164           0.870\n",
       "L2.Technology         0.045806         0.093859            0.488           0.626\n",
       "L3.Office             0.051109         0.037720            1.355           0.175\n",
       "L3.Furniture         -0.092004         0.088573           -1.039           0.299\n",
       "L3.Technology         0.043746         0.090880            0.481           0.630\n",
       "L4.Office             0.003115         0.037874            0.082           0.934\n",
       "L4.Furniture         -0.045003         0.088905           -0.506           0.613\n",
       "L4.Technology        -0.132219         0.091270           -1.449           0.147\n",
       "L5.Office             0.063509         0.037669            1.686           0.092\n",
       "L5.Furniture          0.224865         0.088552            2.539           0.011\n",
       "L5.Technology        -0.145367         0.092335           -1.574           0.115\n",
       "L6.Office            -0.006983         0.038194           -0.183           0.855\n",
       "L6.Furniture          0.046823         0.090226            0.519           0.604\n",
       "L6.Technology        -0.001556         0.093134           -0.017           0.987\n",
       "L7.Office             0.043639         0.036986            1.180           0.238\n",
       "L7.Furniture         -0.018059         0.091727           -0.197           0.844\n",
       "L7.Technology        -0.038457         0.092236           -0.417           0.677\n",
       "================================================================================\n",
       "\n",
       "Correlation matrix of residuals\n",
       "                Office  Furniture  Technology\n",
       "Office        1.000000   0.566179    0.528463\n",
       "Furniture     0.566179   1.000000    0.469971\n",
       "Technology    0.528463   0.469971    1.000000\n",
       "\n"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "grid_searcher.best_forecaster_._fitted_forecaster.summary()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sktime.performance_metrics.forecasting import MeanAbsolutePercentageError\n",
    "from sktime.forecasting.model_evaluation import evaluate\n",
    "\n",
    "mape = MeanAbsolutePercentageError(symmetric=False)\n",
    "\n",
    "cv_results = evaluate(\n",
    "    forecaster=grid_searcher.best_forecaster_,  \n",
    "    y=data,\n",
    "    cv=cv,\n",
    "    strategy=\"refit\",\n",
    "    scoring=mape\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.8658879947147138"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "cv_results.test_MeanAbsolutePercentageError[cv_results.test_MeanAbsolutePercentageError < 100].mean()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Как видим, в большинстве случаев модель дает относительно разумные предсказания, но иногда, из-за близости исходных значений к нулю, MAPE принимает экстремально большие значения, которые не имеет смысла принимать по внимание. Так или иначе, сравнивать с другими поделям нужно будем по средней абсолютной ошибке или по средней квадратичной ошибке."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Triple ARIMA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sktime.forecasting.arima import ARIMA\n",
    "from sktime.forecasting.model_selection import SlidingWindowSplitter, ForecastingGridSearchCV\n",
    "from sktime.forecasting.base import ForecastingHorizon\n",
    "from numpy.typing import ArrayLike\n",
    "\n",
    "def grid_searcher_sliding_arima(\n",
    "        data:ArrayLike, \n",
    "        max_p:int=10,\n",
    "        max_q:int=10,\n",
    "        max_d:int=1\n",
    "    ) -> ForecastingGridSearchCV:\n",
    "    fh = ForecastingHorizon(1)\n",
    "\n",
    "    cv = SlidingWindowSplitter(start_with_window=True, fh=fh, step_length=10, window_length=50)\n",
    "    forecaster = ARIMA(enforce_invertibility=False, enforce_stationarity=False, suppress_warnings=True)\n",
    "    arima_orders = []\n",
    "    start_p, start_q, start_d = 1, 1, 0\n",
    "    for p in range(start_p, max_p+1):\n",
    "        for q in range(start_q, max_q+1):\n",
    "            for d in range(start_d, max_d + 1):\n",
    "                if set([p, q]) != set([6, 7]):\n",
    "                    arima_orders.append((p, d, q))\n",
    "    param_grid = {\"order\": arima_orders}\n",
    "\n",
    "    grid_searcher = ForecastingGridSearchCV(\n",
    "        forecaster=forecaster,\n",
    "        cv=cv,\n",
    "        param_grid=param_grid,\n",
    "        error_score='raise',\n",
    "        backend=\"multiprocessing\"\n",
    "    )\n",
    "    data.fillna(0, inplace=True)\n",
    "    grid_searcher.fit(data)\n",
    "    return grid_searcher"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "tech_gs = grid_searcher_sliding_arima(\n",
    "    data=technology,\n",
    "    max_p=10,\n",
    "    max_q=10,\n",
    "    max_d=1\n",
    ")\n",
    "office_gs = grid_searcher_sliding_arima(\n",
    "    data=office,\n",
    "    max_p=10,\n",
    "    max_q=10,\n",
    "    max_d=1\n",
    ")\n",
    "furniture_gs = grid_searcher_sliding_arima(\n",
    "    data=furniture,\n",
    "    max_p=10,\n",
    "    max_q=10,\n",
    "    max_d=1\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    tech_gs.cv_results_.mean_test_MeanAbsolutePercentageError.mean() = 0.5230651681578081,\n",
      "    office_gs.cv_results_.mean_test_MeanAbsolutePercentageError.mean() = 0.404896504077993,\n",
      "    furniture_gs.cv_results_.mean_test_MeanAbsolutePercentageError.mean() = 0.887560828360154\n",
      "    \n"
     ]
    }
   ],
   "source": [
    "print(\n",
    "    f\"\\\n",
    "    {tech_gs.cv_results_.mean_test_MeanAbsolutePercentageError.mean() = },\\n\\\n",
    "    {office_gs.cv_results_.mean_test_MeanAbsolutePercentageError.mean() = },\\n\\\n",
    "    {furniture_gs.cv_results_.mean_test_MeanAbsolutePercentageError.mean() = }\\n\\\n",
    "    \"\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Получилось даже лучше, чем в случае векторной авторегрессии"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": ".venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
